src/facade/dragonfly_connection.cc

```diff
@@ -990,9 +996,23 @@ void Connection::SendAsync(MessageHandle msg) {
         fb2::Fiber(dfly::Launch::post, "connection_dispatch", [&] { DispatchFiber(peer); });
   }
 
+  auto maybe_reorder_dispatch_q = [this](MessageHandle msg) {
+    std::vector<MessageHandle> updates;
+    while (std::holds_alternative<AclUpdateMessage>(dispatch_q_.front().handle)) {
+      updates.push_back(std::move(dispatch_q_.front()));
+      dispatch_q_.pop_front();
+    }
+    updates.push_back(std::move(msg));
+    for (auto elem = updates.rbegin(); elem != updates.rend(); ++elem) {
+      dispatch_q_.push_front(std::move(*elem));
+    }
+  };
+
   dispatch_q_bytes_.fetch_add(msg.UsedMemory(), memory_order_relaxed);
   if (std::holds_alternative<AclUpdateMessage>(msg.handle)) {
-    dispatch_q_.push_front(std::move(msg));
+    // We need to reorder the queue, since multiple updates might happen before we
+    // pop the message, invalidating the correct order since we always push at the front
+    maybe_reorder_dispatch_q(std::move(msg));
```

Comment:
This was a bug if we pushed two messages one after the other before the queue is saturated (because the order would be messed up). This is fixed now :)